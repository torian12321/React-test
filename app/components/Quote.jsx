var React = require('react');

class Quote extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div className="quote">
                { this.props.children }
                <Quote_author author={ this.props.author }/>
            </div>
        );
    }
}
class Quote_author extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        if(this.props.author){
            return(
                <span className="author">
                    { this.props.author }
                </span>
            );
        }
        return null;
    }
}

Quote.propTypes = {
    children: React.PropTypes.string.isRequired,
    author  : React.PropTypes.string
};
Quote_author.propTypes = {
    author: React.PropTypes.string
};

module.exports = Quote;
