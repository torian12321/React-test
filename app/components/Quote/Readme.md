Basic
```
<Quote>Some nice quote</Quote>
```

With author reference
```
<Quote author='Somebody Junior'>Some nice quote</Quote>
```