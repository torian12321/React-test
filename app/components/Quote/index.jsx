var
React     = require('react'),
PropTypes = require('prop-types'),
styles    = require('./_style.less'),
Icon      = require('Icon');

const Quote = props =>(
    <div className="quote">
        {props.children}
        {props.author ? <span className="author">{ props.author }</span> : null}
        <Icon className="icon-open"  name="quote" />
        <Icon className="icon-close" name="quote" rotate={180} />
    </div>
);

Quote.propTypes = {
    children: PropTypes.string.isRequired,
    author  : PropTypes.string
};

module.exports = Quote;
