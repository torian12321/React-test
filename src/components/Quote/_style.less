@import '~LESS_conf';

.quote{
	font-size  : @fontSize_base;
	position   : relative;
	display    : inline-block;
	max-width  : @bp_lg * 0.8;
	padding    : .5em 2.5em;
	text-align : center;
	line-height: 1.75;
	.smart-text-color();
	.font_elegant();
	.author{
		display   : block;
		margin-top: 1em;
		font-size : .8em;
		text-align: right;
		opacity   : .8;
	}
	.icon{
		position : absolute;
		top      : -.2em;
		font-size: 2.5em;
		fill     : @themeColor_2;
		&.icon-open {left : 0;}
		&.icon-close{right: 0;}
	}
}