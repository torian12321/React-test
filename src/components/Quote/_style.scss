@import '../../styles/conf/index.scss';

// Vars
$quote-font-size  : $font-size !default;
$quote-icon-size : .8em !default;
$quote-icon-color: $themeColor_2 !default;

// Block
.quote {
	font-size  : $quote-font-size;
	position   : relative;
	display    : inline-block;
	max-width  : $bp_lg * 0.8;
	padding    : .5em 2.5em;
	text-align : center;
  line-height: 1.75;

  @include smart-text-color();
  @include font-elegant();
  
  // Elements
	&__author {
		display   : block;
		margin-top: 1em;
		font-size : $quote-icon-size;
		text-align: right;
		opacity   : .8;
	}
	&__icon {
		position : absolute;
		top      : -.2em;
		font-size: 2.5em;
    fill     : $quote-icon-color;

    // Element Modifiers
		&--open {left : 0;}
		&--close{right: 0;}
	}
}