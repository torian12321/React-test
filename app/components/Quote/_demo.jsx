import React from 'react'
import Demo, {props as P} from 'react-demo'
import Quote from './index'


module.exports = () => (
	<div>
		<Demo
			target={Quote}
			props ={{
				children: P.string('Some nice quote')
			}}
		/>
		<Demo
			target={Quote}
			props ={{
				children: P.string('Some nice quote'),
				author  : P.string('Lady Gaga')
			}}
		/>
	</div>
);