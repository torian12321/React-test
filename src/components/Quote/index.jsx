import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import Icon       from '../Icon';
import './_style.less';

class Quote extends React.PureComponent {
  render(){
    return(
      <blockquote className={classNames('quote', this.props.className)} cite={this.props.cite}>
        {this.props.children}
        {this.props.author ? <span className="quote-author">{ this.props.author }</span> : null}
        <Icon className="icon-open"  name="quote" />
        <Icon className="icon-close" name="quote" rotate={180} />
      </blockquote>
    );
  }
}

Quote.propTypes = {
    children : PropTypes.string.isRequired,
    className: PropTypes.string,
    author   : PropTypes.string,
    cite     : PropTypes.string
};

module.exports = Quote;
