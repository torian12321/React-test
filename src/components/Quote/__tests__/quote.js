import React    from 'react';
import renderer from 'react-test-renderer';
import Quote    from '../index';

describe('component', () => {
  describe('<Quote />', () => {
    
    it('should render correctly', () => {
      expect(renderer.create(
        <Quote>Lorem ipsum</Quote>
      )).toMatchSnapshot();
    });

    it('should render Autor and Cite', () => {
      expect(renderer.create(
        <Quote author='someone' cite='somewhere'>Lorem ipsum</Quote>
      )).toMatchSnapshot();
    });

  });
});
