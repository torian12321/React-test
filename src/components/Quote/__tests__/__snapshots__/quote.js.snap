// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Quote /> should render Autor and Cite 1`] = `
<blockquote
  cite="somewhere"
  className="quote"
>
  Lorem ipsum
  <span
    className="quote-author"
  >
    someone
  </span>
  <svg
    className="icon icon-open"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 16v8h8v-8H4s0-4 4-4V8s-8 0-8 8zm20-4V8s-8 0-8 8v8h8v-8h-4s0-4 4-4z"
      />
    </g>
  </svg>
  <svg
    className="icon icon-close"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": "rotate(180deg)",
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 16v8h8v-8H4s0-4 4-4V8s-8 0-8 8zm20-4V8s-8 0-8 8v8h8v-8h-4s0-4 4-4z"
      />
    </g>
  </svg>
</blockquote>
`;

exports[`component <Quote /> should render correctly 1`] = `
<blockquote
  cite={undefined}
  className="quote"
>
  Lorem ipsum
  <svg
    className="icon icon-open"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": null,
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 16v8h8v-8H4s0-4 4-4V8s-8 0-8 8zm20-4V8s-8 0-8 8v8h8v-8h-4s0-4 4-4z"
      />
    </g>
  </svg>
  <svg
    className="icon icon-close"
    style={
      Object {
        "fill": null,
        "height": null,
        "transform": "rotate(180deg)",
        "width": "1em",
      }
    }
    viewBox="0 0 32 32"
  >
    <g>
      <path
        d="M0 16v8h8v-8H4s0-4 4-4V8s-8 0-8 8zm20-4V8s-8 0-8 8v8h8v-8h-4s0-4 4-4z"
      />
    </g>
  </svg>
</blockquote>
`;
